sound/soc/sof/imx/imx8.c

```diff
@@ -116,7 +117,16 @@ static void imx8_dsp_handle_request(struct imx_dsp_ipc *ipc)
 {
 	struct imx8_priv *priv = imx_dsp_get_data(ipc);
 
-	snd_sof_ipc_msgs_rx(priv->sdev);
+	int p; /* panic code */
+
+	/* Read the message from the debug box. */
+	sof_mailbox_read(priv->sdev, priv->sdev->debug_box.offset + 4, &p, 4);
+
+	/* Check to see if the message is a panic code (0x0dead***) */
+	if ((p & SOF_IPC_PANIC_MAGIC_MASK) == SOF_IPC_PANIC_MAGIC)
```

Comment:
Here we should go with a comment like this: ```/* Handle messages from DSP Core */```

Most of the time there will be no panics.